src/platform/nrfconnect/FactoryDataParser.c

```diff
@@ -84,6 +104,105 @@ static bool DecodeEntry(zcbor_state_t * states, void * buffer, size_t bufferSize
     return res;
 }
 
+bool SerializeFactoryData(uint8_t * factoryDataBuff, size_t factoryDataSize, const struct FactoryData * factoryData)
+{
+    zcbor_state_t cbor_state[MAX_FACTORY_DATA_NESTING_LEVEL];
+    zcbor_new_encode_state(cbor_state, MAX_FACTORY_DATA_NESTING_LEVEL, factoryDataBuff, factoryDataSize, 0);
+
+    if (!(zcbor_map_start_encode(cbor_state, 1) && zcbor_tstr_put_lit_cast(cbor_state, "version") &&
```

Comment:
I'd think of a private helper function like DoSerialize which would wrap lines 112-142 and here only call `VerifyOrReturnValue(DoSerialize(), false)` or similar.

Reply:
edit: valid only if we keep the serialization instead of using factory data with invalidated DAC.

Reply:
Serialization will be removed.